dev/io.openliberty.security.jakartasec.3.0.internal/src/io/openliberty/security/jakartasec/identitystore/OidcIdentityStore.java

```diff
@@ -72,7 +74,7 @@ public CredentialValidationResult validate(Credential credential) {
 
                     OidcClientConfig oidcClientConfig = client.getOidcClientConfig();
                     long tokenMinValidityInMillis = oidcClientConfig.getTokenMinValidity();
-                    AccessToken accessToken = createAccessTokenFromTokenResponse(tokenMinValidityInMillis, tokenResponse);
+                    AccessToken accessToken = createAccessTokenFromTokenResponse(tokenMinValidityInMillis, tokenResponse, idTokenClaims);
```

Comment:
The access token string from tokenResponse.getAccessTokenString() should be used to get the access token claims map to use instead of the idTokenClaims. The createAccessTokenFromTokenResponse method could be enhanced to parse the access token string.

Reply:
Redid how we decide on if it's a jwt token.